### PR TITLE
[NEXUS-8291] Add support for ?describe={html|json}

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8291

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF195-1 :white_check_mark:

**Comment by adreghiciu** [comment]
+1

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescribeType.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * @since 3.0
+ */
+public enum DescribeType
+{
+  HTML,
+  JSON;
+
+  /**
+   * Parse type from flags.  Either explicit HTML or JSON, or anything else will default to HTML.
```
Comment (peterlynch) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescribeType.java]: inaccurate. null will throw, and value is trimmed and made case insensitive by this method before explicit match.

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/ViewServlet.java [hunk 2]
```diff
@@ -193,7 +196,21 @@ Response describe(final Request request, final Response response, final Exceptio
...
-    return toHtmlResponse(description);
+
+    DescribeType type = DescribeType.parse(flags);
+    log.trace("Describe type: {}", type);
+    switch (type) {
+      case HTML: {
+        String html = descriptionRenderer.renderHtml(description);
+        return HttpResponses.ok(new StringPayload(html, Charsets.UTF_8, "text/html"));
+      }
+      case JSON: {
```
Comment (peterlynch) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/ViewServlet.java]: personally would have added test to verify content types and JSON payloads, and status codes, including for bad describe param

**Comment by peterlynch** [comment]
I think this is good direction as is using a parameter value. Would still eventually like to see additional dynamic response type selection as is typical in REST-like API.

**Comment by cstamas** [comment]
+1

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/ViewServlet.java (hunk 2 above)
Comment (jdillon) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/ViewServlet.java]: there are no bad describe params.
and tests can be added.

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescribeType.java (hunk 1 above)
Comment (jdillon) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescribeType.java]: whiny whiny :-P

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescribeType.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   */
+  public static DescribeType parse(String flags) {
+    checkNotNull(flags);
+    flags = flags.toUpperCase().trim();
```
Comment (mcculls) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescribeType.java]: Minor: in general `toUpperCase(Locale.ENGLISH)` is preferred over using the platform locale

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescriptionItem.java [hunk 4]
```diff
@@ -23,9 +23,15 @@
 
   private final String type;
 
+  /**
+   * Value should be a primative/char-sequence or collection/map of primative/char-sequence
```
Comment (mcculls) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescriptionItem.java]: Spelling: primitive ;)

**Comment by mcculls** [comment]
+1 couple of very minor comments

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**